UnitTests/Sources/PermalinkBuilderTests.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@testable import ElementX
+import XCTest
+
+class PermalinkBuilderTests: XCTestCase {
+    func testUserIdentifierPermalink() {
+        let userId = "@abcdefghijklmnopqrstuvwxyz1234567890._-=/:matrix.org"
+        
+        do {
+            let permalink = try PermalinkBuilder.permalinkTo(userIdentifier: userId)
+            XCTAssertEqual(permalink, URL(string: "\(BuildSettings.matrixDotToUrl)/#/\(userId)"))
+        } catch {
+            XCTFail("User identifier must be valid: \(error)")
+        }
+    }
+    
+    func testInvalidUserIdentifier() {
+        do {
+            _ = try PermalinkBuilder.permalinkTo(userIdentifier: "This1sN0tV4lid!@#$%^&*()")
```

Comment:
Same comment for all `testInvalid…`:
```suggestion
            _ = try PermalinkBuilder.permalinkTo(userIdentifier: "This1sN0tV4lid!@#$%^&*()")
            XCTFail("A permalink should not be created.")
```